broker/src/tunneldigger_broker/netlink.py

```diff
@@ -173,12 +181,8 @@ def recv(self, multiple = False):
                     done = True
                     break
                 elif msg.type == NLMSG_ERROR:
-                    errno = -struct.unpack("i", msg.payload[:4])[0]
-                    if errno != 0:
-                        err = OSError("Netlink error: %s (%d)" % (
-                                                             os.strerror(errno), errno))
-                        err.errno = errno
-                        raise err
+                    # an ack
```

Comment:
This is only possible in case the `errno` contained in the message is zero. Currently, the handling in this case is inconsistent when `multiple` is `True` (error message is skipped and not included in the final list) and when `multiple` is `False` (error message is appended to the list of received messages). I think this should be unified.

Reply:
When `multiple` is `False`, the entire point of calling this message is to receive the ACK. If we skip it, all uses of `multiple=False` will break.

Maybe it makes more sense to have a dedicated function `recv_ack`, and rename this one into `recv_until_done`?

Reply:
I see, so an ACK is actually a message of type `NLMSG_ERROR`, but with an errno payload of zero. The problem with two methods is that they would most likely be very similar, so perhaps it would be better to just document this behavior in the method's documentation block.

This whole netlink code is really ugly as it was imported from somewhere else and we should eventually refactor it.

Reply:
> an ACK is actually a message of type NLMSG_ERROR, but with an errno payload of zero.

Correct.

> better to just document this behavior in the method's documentation block.

Will do.